SuperBuild/External_SlicerExecutionModel.cmake

```diff
@@ -56,9 +56,14 @@ if(NOT DEFINED SlicerExecutionModel_DIR AND NOT ${CMAKE_PROJECT_NAME}_USE_SYSTEM
     ${Slicer_INSTALL_CLIMODULES_LIB_DIR})
 
   if(Slicer_BUILD_PARAMETERSERIALIZER_SUPPORT)
-    _set(JsonCpp_INCLUDE_DIR PATH ${JsonCpp_INCLUDE_DIR})
-    _set(JsonCpp_LIBRARY PATH ${JsonCpp_LIBRARY})
     _set(ParameterSerializer_DIR PATH ${ParameterSerializer_DIR})
+    _set(JsonCpp_INCLUDE_DIR PATH ${JsonCpp_INCLUDE_DIR})
+
+    # JsoncCpp_LIBRARY needs to be added as a CMAKE_ARGS because it contains an
+    # expression that needs to be evaluated
+    list(APPEND EXTERNAL_PROJECT_OPTIONAL_ARGS
+      CMAKE_ARGS -DJsonCpp_LIBRARY:PATH=${JsonCpp_LIBRARY}
+      )
```

Comment:
Nice. This will indeed fix the problem.

But I would prefer that you:
1. Rename `EXTERNAL_PROJECT_OPTIONAL_ARGS` to `EXTERNAL_PROJECT_OPTIONAL_CMAKE_CACHE_ARGS` like it is done in `External_ITKv4.cmake`
2. set `EXTERNAL_PROJECT_OPTIONAL_CMAKE_ARGS` to an empty string outside of the `if`, append optonal args to it. Then, just before the call to `ExternalProject_Add`, you woud do:

```
# Note: Remove this block if non optional CMAKE_ARGS are added to ExternalProject_Add.
if(NOT EXTERNAL_PROJECT_OPTIONAL_CMAKE_ARGS STREQUAL "")
  set(EXTERNAL_PROJECT_OPTIONAL_CMAKE_ARGS
    CMAKE_ARGS
    ${EXTERNAL_PROJECT_OPTIONAL_CMAKE_ARGS})
endif()
```

Doing so will make the code easier to maintain.

Thanks